tools/kmodules/sched_tp/sched_events.h

```diff
@@ -297,6 +279,33 @@ TRACE_EVENT_CONDITION(uclamp_util_cfs,
 #define trace_uclamp_util_cfs_enabled() false
 #endif /* CONFIG_UCLAMP_TASK */
 
+TRACE_EVENT(sched_cpu_capacity,
+
+	TP_PROTO(struct rq *rq),
+
+	TP_ARGS(rq),
+
+	TP_STRUCT__entry(
+		__field(	int,		cpu		)
+		__field(	unsigned long,	capacity	)
+		__field(	unsigned long,	capacity_orig	)
+		__field(	unsigned long,	capacity_curr	)
+	),
+
+	unsigned long scale_cpu = arch_scale_cpu_capacity(rq->cpu);
+	unsigned long scale_freq = arch_scale_freq_capacity(rq->cpu);
+
+	TP_fast_assign(
+		__entry->cpu		= rq->cpu;
```

Comment:
We used sched_tp_rq_cpu() because rq could be NULL on some of the call sites. I assume we don't have this problem here? I don't mind it, just asking to make sure we thought about it :)